raster-foundry/raster-foundry
# Add taskId to annotations

## Overview

This PR adds `taskId` to annotations so that they can be explicitly linked to tasks. This also adds `taskId` as a query parameter.

### Checklist


### Notes

There was a `task` QP in the spec that wasn't implemented so I updated it to reflect the new name of the QP.

## Testing Instructions

- If you have tasks in your dev db, grab an id from one.
- If you don't have tasks in your dev db, drop the foreign key constraint: `alter table annotations drop constraint "annotations_task_id_fkey";`
- Create some annotations if you don't have any in your db
- Update annotations with the task id from a task
- Send some requests with the `taskId` param (test with non-existent IDs too) and see that it returns what you'd expect

Closes #5072
